perception/autoware_image_projection_based_fusion/src/pointpainting_fusion/node.cpp

```diff
@@ -338,12 +347,13 @@ dc   | dc dc dc  dc ||zc|
     p_y = point_camera.y();
     p_z = point_camera.z();
 
-    if (p_z <= 0.0 || p_x > (tan_h_.at(image_id) * p_z) || p_x < (-tan_h_.at(image_id) * p_z)) {
-      continue;
-    }
+    /// check if the point is in the camera view
+    if (p_z <= 0.0) continue;
     // project
     Eigen::Vector2d projected_point = calcRawImageProjectedPoint(
       pinhole_camera_model, cv::Point3d(p_x, p_y, p_z), point_project_to_unrectified_image_);
+    if (projected_point.x() < fov_left_.at(image_id)) continue;
+    if (projected_point.x() > fov_right_.at(image_id)) continue;
```

Comment:
I think this part is wrong since the `projected_point` is projected onto the image space.
After projecting the points onto the image plane, the check should be whether it is within [0, width-1].

So if we don't check before the projection, we don't need the `tan_h_ (fov_left_, fov_right_)` anymore, I guess.

Reply:
https://github.com/autowarefoundation/autoware.universe/blob/932e87fde0614e9342e8889c7961f103e615e4e3/perception/autoware_image_projection_based_fusion/src/pointpainting_fusion/node.cpp#L287-L298

`lidar2cam_affine` is transformation matrix between the frames. the target frame of the camera is to optic center. Therefore, it is not the image space but the camera coordinate system.

It can be checked experimentally by replace `fov_right_.at(image_id)` to zero. if there are points overwritten, then the projected coordinate is not in [0, width-1].

Reply:
Sorry, which part are you talking about?
I agree about the `lidar2cam_affine` and the `point_camera`.
https://github.com/autowarefoundation/autoware.universe/blob/932e87fde0614e9342e8889c7961f103e615e4e3/perception/autoware_image_projection_based_fusion/src/pointpainting_fusion/node.cpp#L344

But the `projected_point` is on the image space (pixel coordinate) right?, or am I misunderstanding somethings?
https://github.com/autowarefoundation/autoware.universe/blob/932e87fde0614e9342e8889c7961f103e615e4e3/perception/autoware_image_projection_based_fusion/src/pointpainting_fusion/node.cpp#L353-L354

Reply:
Oh, project3dToPixel adds the optical center... Your understand is correct.

Reply:
resolved by https://github.com/autowarefoundation/autoware.universe/pull/9637/commits/355c183c4b549dac094a59483f0e2f3169dc5149